src/common-ui/Text.test.tsx

```diff
@@ -0,0 +1,124 @@
+import { render, screen } from '@testing-library/react';
+import Text from './Text';
+
+describe('Text test', () => {
+  test('Text 컴포넌트는 children을 렌더링한다.', () => {
+    //given
+    const TEXT = 'Hello World';
+
+    //when
+    render(
+      <Text type={{ tag: 'h1', type: 'header' }} weight="bold">
+        {TEXT}
+      </Text>,
+    );
+
+    //then
+    expect(screen.getByText(TEXT)).toBeInTheDocument();
+  });
+  test('Text 컴포넌트는 type에 따라 태그를 렌더링한다.', () => {
+    //given
+    const TEXT = 'Hello World';
+
+    //when
+    render(
+      <Text type={{ tag: 'h2', type: 'header' }} weight="bold">
+        {TEXT}
+      </Text>,
+    );
+
+    //then
+    expect(screen.getByRole('heading', { level: 2 })).toBeInTheDocument();
+  });
+  test('Text 컴포넌트는 weight에 따라 폰트를 렌더링한다. Bold', () => {
+    //given
+    const TEXT = 'Hello World';
+
+    //when
+    render(
+      <Text type={{ tag: 'h2', type: 'header' }} weight="bold">
+        {TEXT}
+      </Text>,
+    );
+
+    //then
+    expect(screen.getByText(TEXT)).toHaveStyle(
+      'font-family: NanumSquareRoundB',
+    );
+  });
```

Comment:
여기도 마찬가지로 given절에 weight 가 있으면 테스트가 더 명확해질 것 같습니다.
```suggestion
  test('Text 컴포넌트는 weight에 따라 폰트를 렌더링한다. Bold', () => {
    //given
    const FONT_WEIGHT = 'bold';

    //when
    render(
      <Text type={{ tag: 'h2', type: 'header' }} weight={FONT_WEIGHT}>
         Text
      </Text>,
    );

    //then
    expect(screen.getByText(TEXT)).toHaveStyle(
      'font-family: NanumSquareRoundB',
    );
  });
```